schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+    option (google.api.http) = {
+      patch: "/v1alpha3/{name=rooms/*}"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name", "room.display_name"]
+      additional_signatures: {
+        fields: ["name", "room.display_name", "room.description"]
+      }
+    };
+  }
+
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  rpc StreamRoom(StreamRoomRequest) returns (stream Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{name=rooms/*}:streamRoom"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{name=global}:streamRoom"
+        body: "*"
+      }
+    };
+  }
+
+  rpc SearchAll(SearchAllRequest) returns (google.longrunning.Operation) {
+    option (google.api.http) = {
+      post: "/v1alpha3:searchAll"
+      body: "*"
+    };
+    option (google.api.operation) = {
+      response_type: "google.showcase.v1alpha3.SearchAllResponse"
+      metadata_type: "google.showcase.v1alpha3.SearchAllMetadata"
+    };
+    option (google.api.method_signature) = {
+      fields: ["query"]
+    };
+  }
+
+  rpc CreateMessage(CreateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/messages"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{parent=global}/messages",
+        body: "*"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "message"]
+    };
+  }
+
+  rpc GetMessage(GetMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/name={messages/*}"
+      }
+    };
+  }
+
+  rpc UpdateMessage(UpdateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      body: "*"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc DeleteMessage(DeleteMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc ListMessages(ListMessagesRequest) returns (ListMessagesResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/messages"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/messages"
+      }
+    };
+  }
+
+  rpc StreamMessages(stream CreateMessageRequest) returns (google.protobuf.Empty) {
```

Comment:
To clarify the semantics: What does `StreamMessages` stream to the server?  A series of messages that were queued up because the user was offline? (Trying to see the real-world use case...)

How about streaming an image file being uploaded by the client to the server?

Reply:
Hm. I think streaming an Image file could be a good alternative, let me try and improve upon what I have, before switching to a method with a different behavior.

Reply:
`StreamMessages` makes me expect to stream messages _down_. I might suggest something like "send messages"

I agree with @vchudnov-g that this probably would not be a real RPC if we actually wanted to run this service for more than a testbed, but I do not think it matters provided the RPC has an intuitive name. What matters is whether it will cause cognitive overhead for a reader.

If you do want to tweak it, perhaps streaming _audio_ would be a more sensible example? (e.g.  you say something into the API and it transcribes it) Images are too small to need streaming the bulk of the time.

Reply:
I changed it to `SendMessages`